### PR TITLE
ROX-20058: Use GCP secret manager + Helm for infra deployments

### PR DESCRIPTION
**Pull request by tommartensen**
# Changes

- Infra configuration and secrets were moved to GCP secret manager, which supports versioning, streaming of values into `helm upgrade`.
- Infra is now deployed via Helm.
- Helm and secrets utils are available through Make targets, which use the scripts in `scripts/deploy`.
- "deployment" & "environment" were simplified to "environment", which can be "development" or "production". Previous usages of "local" deployment have been replaced with check of the `.Capabilities` API to avoid provisioning non-existent K8s resources, and the `testMode` value, which can be set to disable telemetry. 
- The "infra" namespace is now created with Helm instead of explicitly as a K8s resource

## Misc
- PR GHA workflow: Is now synchronized, ie only one workflow per PR can run
- Deployment GHA workflow: The Slack channel ID was fixed
- Documentation was updated
- Argo Workflow server is now installed through a Helm chart and independent of the infra-server deployment 

# How I tested my changes

- PR workflow (build, PR cluster, e2e tests)
  - Port-forward to PR cluster, login through UI on https://localhost:8443
- Deployed to dev.infra.rox.systems with the fixed Deployment workflow ([notification](https://redhat-internal.slack.com/archives/CVANK5K5W/p1697026997469119)) and from my machine, following the DEPLOYMENT.md instructions.

# Rollout on prod

1. `make configuration-download` on master
2. Replacing ".Values.deployment" with ".Values.environment" in chart/infra-server/configuration/production/{auth0.yaml, oidc.yaml}
3. `make create-consolidated-values`
4. Validate "oidc_yaml" contains ".Values.environment":

    ```bash
    yq \
        ".oidc_yaml" \
        "chart/infra-server/configuration/production-values-from-files.yaml" \
    | base64 --decode
    ```

5. Remove "auth0_yaml" (migrated to RHSSO), "google_calendar_credentials_json" (functionality removed), "gke__gke_credentials_json" (pre RH migration), all unused in chart.
6. Switch to `tm/helm-charts` branch.
7. `ENVIRONMENT=production make secrets-upload`
8. Validate new secret shows updated values: `./scripts/deploy/secrets.sh show production latest`, selecting "oidc_yaml" shows the same output as (4).
9. Validate you're pointing to the correct cluster: `kubectl config current-context`.
10. Delete argo and infra namespaces and all secrets in default namespace (except default token). This is required to have a clean slate, otherwise conflicts with Helm unmanaged resources.
11. `ENVIRONMENT=production make install-argo helm-deploy`.

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -53,14 +54,14 @@ jobs:
...
-          make install-${{ inputs.environment }}
+          ENVIRONMENT=${{ inputs.environment }} make install-argo helm-deploy
 
       - name: Notify infra channel about new version
         env:
           SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
         uses: slackapi/slack-github-action@v1.23.0
         with:
-          channel-id: C01H4DC33K3 #acs-infra
+          channel-id: CVANK5K5W #acs-infra
```
Comment (gavin-stackrox) [inline .github/workflows/deploy.yaml:64]: Yay!

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+name: Deploy infra
```
Comment (gavin-stackrox) [inline .github/workflows/deploy.yaml:1]: Yay!

File: Makefile [hunk 3]
```diff
@@ -23,92 +30,6 @@ IMAGE=us.gcr.io/stackrox-infra/infra-server:$(VERSION)
 image-name:
 	@echo $(IMAGE)
 
-#############
-## Linting ##
-#############
```
Comment (gavin-stackrox) [inline Makefile:28]: Ugh. Re-organizing files makes review extra hard. Re-org _should_ be a separate PR IMhO. But not going to hold this PR back for it.

File: chart/infra-server/templates/certificate.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-{{ if ne .Values.deployment "local" -}}
+{{- if .Capabilities.APIVersions.Has "networking.gke.io/v1/ManagedCertificate" -}}
```
Comment (gavin-stackrox) [inline chart/infra-server/templates/certificate.yaml:1]: This is going to create the managedcertificate on infra PR instances with dev values. How will this affect the dev instance? If we are not sure, then best to avoid.

File: chart/infra-server/templates/ingress.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{{ if ne .Values.deployment "local" -}}
+{{- if .Capabilities.APIVersions.Has "networking.gke.io/v1/Ingress" -}}
```
Comment (gavin-stackrox) [inline chart/infra-server/templates/ingress.yaml:1]: Same concern as regards managedcertificate.

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]
Nice! Other than my questions around ManagedCert/Ingress, I think this PR is good to go

File: chart/infra-server/templates/certificate.yaml (hunk 4 above)
Comment (tommartensen) [reply chart/infra-server/templates/certificate.yaml:1]: The managed certificate will be stuck in Provisioning state for infra PR instances, because the certificate challenge cannot be reached on the `{{ .Values.hosts.primary }}` or `{{ .Values.hosts.secondary }}`.

No impact on the dev cluster other than some failed requests from GCP.

Comment (tommartensen) [reply chart/infra-server/templates/certificate.yaml]: OK https://github.com/stackrox/infra/pull/1015/commits/ce8e555ae19ce0136a884d69bf213289654fffc0 :

**Review by gavin-stackrox — APPROVED** [review]
